eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala

```diff
@@ -487,6 +487,47 @@ object PaymentRequest {
     )
   }
 
+  private def readBoltData(input: String): Bolt11Data = {
```

Comment:
Since you're still decoding the data part, I'm honestly unsure that this is faster than completely decoding the invoice (now that we removed signature validation from it). Do you have some performance measures on android?
I'll do some testing on a laptop and will post my numbers here (if this isn't more performant, it's less error-prone to simply decode the invoice).